plugins/dynamix/FTP.page

```diff
@@ -16,39 +16,142 @@ Tag="globe"
+$is_ftp_syslog_enabled = preg_match('/syslog_enable\s*=\s*yes/i', exec("grep syslog_enable /etc/vsftpd.conf"));
 ?>
+
 <script>
 $(function() {
   showStatus('21');
 });
+
+function updateFTPPermissions(form)
+{
+  var permissions = [];
+  
+  $(form).find(".ftp_permissions input[type='text']").each(function(){
+    var value = this.value;
+    var name = this.name;
+    permissions.push(`${name}=${value}`);
+  });
+
+  $(form).find('input[name="#arg[1]"]').val(permissions.join("|"));
+
+  return true;
+}
+
 </script>
 
+:ftp_overview_help:
+
 <form markdown="1" method="POST" action="/update.php" target="progressFrame">
-<input type="hidden" name="#command" value="/webGui/scripts/ftpusers">
+<input type="hidden" name="#command" value="/webGui/scripts/ftpconfiguration">
 
 _(FTP server)_:
 : <select name="#arg[1]">
-  <?=mk_option($ftp_server, "0", _("Disabled"))?>
-  <?=mk_option($ftp_server, "1", _("Enabled"))?>
+  <?=mk_option($is_ftp_server_running, "0", _("Disabled"))?>
+  <?=mk_option($is_ftp_server_running, "1", _("Enabled"))?>
   </select>
 
 :ftp_server_help:
 
-_(FTP user(s))_:
-: <input type="text" name="#arg[2]" size="40" maxlength="80" value="<?=htmlspecialchars($ftp_userlist)?>">
+_(Log to Syslog)_:
+: <select name="#arg[2]">
+  <?=mk_option($is_ftp_syslog_enabled, "0", _("Disabled"))?>
+  <?=mk_option($is_ftp_syslog_enabled, "1", _("Enabled"))?>
+  </select>
 
-:ftp_users_help:
+:ftp_syslog_enable_help:
 
 &nbsp;
 : <input type="submit" value="_(Apply)_" disabled><input type="button" value="_(Done)_" onclick="done()">
 
 </form>
 
-:ftp_overview_help:
+<form markdown="1" method="POST" action="/update.php" target="progressFrame" onsubmit="return updateFTPPermissions(this)">
+<input type="hidden" name="#command" value="/webGui/scripts/ftpusers">
+<input type="hidden" name="#arg[1]" value="root=test">
+
+<?php
+
+echo "<table class='ftp_permissions' style='margin-top:20px'><thead><tr><td>" . _('User') . "</td><td>" . _('Current Root Folder') . "</td><td>" . _('New Root Folder') . "</td></tr></thead><tbody>";
+
+// go through every user
+foreach($users as $user)
+{
+  $user_config_file = $user_config_dir . "/" . $user["name"];
+  $user_root_folder = "";
+  
+  // if the user has a config file
+  if(file_exists($user_config_file) && is_file($user_config_file))
+  {
+    // get the root folder from the config file
+    $user_config = trim(file_get_contents($user_config_file));
+    $user_config_parts = explode("=", $user_config);
+    $user_root_folder = $user_config_parts[1];
+  }
+
+  // show on the screen
+  echo "<tr><td>" . $user["name"] . "</td><td>" . $user_root_folder . "</td><td><input type='text' name='" . $user["name"] . "' value='" . $user_root_folder . "'></td></tr>";
+}
+
+echo "</tbody></table>";
+
+?>
+
+&nbsp;
+: <input type="submit" value="_(Apply)_" disabled><input type="button" value="_(Done)_" onclick="done()">
+</form>
+
+<?PHP
+
+# copied from plugins\dynamix.docker.manager\include\CreateDocker.php
```

Comment:
Unneeded comment.

Reply:
I don't know enough about the core Unraid system and wanted to make sure folks know I just copied this from elsewhere. I was hoping it was a central library I could just reference instead of copying but since it isn't I wanted to make sure it was clear. I defer to y'all on if you do or don't want it. I'm just trying to help by adding a missing feature I think will help.

Reply:
How exactly do I do this now? Do I have to change on my end and submit a new pull request?

Reply:
Pushing a new commit should be fine.

Reply:
Done. Thanks!